src/shape/java/us/ihmc/euclid/shape/convexPolytope/interfaces/ConvexPolytope3DReadOnly.java

```diff
@@ -154,6 +156,52 @@ default void getBoundingBox(BoundingBox3DBasics boundingBoxToPack)
    double getConstructionEpsilon();
 
    /** {@inheritDoc} */
+   
+   
+   
+   @Override
+
+   default int intersectionWith(Point3DReadOnly pointOnLine,
+                                Vector3DReadOnly lineDirection,
+                                Point3DBasics firstIntersectionToPack,
+                                Point3DBasics secondIntersectionToPack)
+   
+   {
+      
+      if (firstIntersectionToPack != null)
+         firstIntersectionToPack.setToNaN();
+      if (secondIntersectionToPack != null)
+         secondIntersectionToPack.setToNaN();
+      
+      int numberOfIntersections = 0;
+      
+      //For each face, we check the possibility of intersection with the line
+      for (int i=0; i< getNumberOfFaces(); i++) {
+         //If there isn't already a intersection found
```

Comment:
The conditions in this loop aren't working, also you've got a bug due to a bad copy-paste. I'll let you figure out how to fix this loop.